### PR TITLE
Flag tweaks

### PR DESCRIPTION
**Pull request by Citinited**
This PR does the following:
- You can now roll up a flag by activating it in your hand. Chameleon flags won't have this ability while undisguised.
- Fixes an issue where you could determine if a flag was a chameleon flag or not.
- Adds some more messages to flag stuff, and gives the `obj/item/flag` an actual description.
- Adds the rest of the post to the kidan flag, as it was missing before.

**Obligatory pictures!**

These were already existing assets in the codebase so it seemed like a shame not to use them.

:cl:
tweak: You can now roll up flags.
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
my god

**Comment by Citinited** [comment]
I'm not a god @FreeStylaLT but thanks anyway

File: code/game/objects/items/flag.dm [hunk 1]
```diff
@@ -23,11 +31,16 @@
 
 /obj/item/flag/update_icon()
 	overlays.Cut()
+	icon_state = replacetext(icon_state, "_rolled", "") //This will totally not break in the future
```
Comment (Citinited) [inline code/game/objects/items/flag.dm]: There's a better way of doing this, will play around with it tomorrow when I'm not as sleepy

Comment (KasparoVy) [reply code/game/objects/items/flag.dm]: icon_state = initial(icon_state)?

Comment (Citinited) [reply code/game/objects/items/flag.dm]: Unfortunately that breaks chameleon flags, current workaround is to have a second var that is set to `icon_state`, and use that for the icon

Comment (KasparoVy) [reply code/game/objects/items/flag.dm]: Fair enough, I totally forgot about the chams
Hearts work the same way

**Comment by Citinited** [comment]
Updated to a better way of doing it

File: code/game/objects/items/flag.dm [hunk 2]
```diff
@@ -23,11 +31,20 @@
 
 /obj/item/flag/update_icon()
 	overlays.Cut()
+	if(istype(src, /obj/item/flag/chameleon))
```
Comment (Crazylemon64) [inline code/game/objects/items/flag.dm]: yikes
`istype(src, ` is not good

Perhaps do this?
```
/obj/item/flag/proc/reset_icon()
  icon_state = initial(icon_state)

/obj/item/flag/chameleon/reset_icon()
  icon_state = C.updated_icon_state
```
And then just call `reset_icon()` in place of this `istype` block here

Comment (Citinited) [reply code/game/objects/items/flag.dm]: Oh, that's a much better way of doing it (I couldn't think of a better way at the time XD). Will update shortly

**Comment by Citinited** [comment]
@Crazylemon64 Updated

**Pull request merged**